test/test_sparse.py

```diff
@@ -509,8 +565,121 @@ def _test_sparse_mask_fixed(self):
         expected = self.SparseTensor(i, exp_v, torch.Size([5, 4]))
         self.assertEqual(res, expected)
 
+        expected_dense = self.ValueTensor([
+            [1, 2, 300, 4],
+            [5, 6, 100, 8],
+            [9, 10, 11, 12],
+            [13, 200, 15, 16],
+            [17, 18, 19, 400],
+        ])
+        dense._sparse_copy_(x)
+        self.assertEqual(dense, expected_dense)
+
+    def prop_sparse_mask_const(self, x, s):
+        """sparse_mask does not mutate inputs."""
+        expected_x = x.clone()
+        expected_s = s.clone()
+        try:
+            x._sparse_mask(s)
+        except:
+            pass
+        self.assertEqual(x, expected_x)
+        self.assertEqual(s, expected_s)
+
+    def _sparse_mask_py(self, x, s):
+        """Python reference implementation of sparse_mask"""
+        if not s.is_coalesced():
+            raise ValueError("mask is not coalesced")
+        if x.size() != s.size():
+            raise ValueError("sparse_mask operands have incompatible sizes")
+        ss = SparseSize(size=x.size(), dimI=s._indices().size()[0], nnz=s._nnz())
+        r = ss.new_sparse(dense2sparse(type(x)), s._indices())
+        v = r._values()
+        for i, js in enumerate(s._indices().t()):
+            m = x
+            for j in js:
+                # NB: m[j] will happily accept a negative index; but sparse_mask
+                # really shouldnot!
+                if j < 0:
+                    raise IndexError("negative index is out of range")
+                m = m[j]
+            v[i] = m
```

Comment:
You are right! Well, technically, it's equivalent to `v[i] = x[tuple(js)]`, because `js` is a tensor and that means it has different semantics.